src/mscorlib/shared/System/IO/PathHelper.Windows.cs

```diff
@@ -42,14 +42,31 @@ internal static string Normalize(string path)
             return result;
         }
 
-        private static void GetFullPathName(string path, ref ValueStringBuilder builder)
+        internal static string Normalize(ReadOnlySpan<char> path)
+        {
+            Span<char> initialBuffer = stackalloc char[PathInternal.MaxShortPath];
+            ValueStringBuilder builder = new ValueStringBuilder(initialBuffer);
+
+            // Get the full path
+            GetFullPathName(path, ref builder);
+
+            string result = builder.AsSpan().Contains('~')
+                ? TryExpandShortFileName(ref builder, originalPath: null)
+                : builder.ToString();
+
+            // Clear the buffer
+            builder.Dispose();
```

Comment:
any reason you don't use a using statement

Reply:
same elsewhere for ValueStringBuilder.

Reply:
Can't. It is a ref struct. :/ Per prior feedback we're ok with not returning buffers to the pool on the error case.

Reply:
Ah of course, cannot assign to interface type.
https://docs.microsoft.com/en-us/dotnet/csharp/reference-semantics-with-value-types#ref-struct-type